src/mixins/transparentize.js

```diff
@@ -0,0 +1,57 @@
+// @flow
+import { isRgb, isRgba, isHsl, isHsla, isHex } from '../internalHelpers/_isValidColor'
+import hexToRgb from '../internalHelpers/_hexToRgb'
+
+const getNewTransparency = (currentTransparency: number, percentage: number) : number => (
+  currentTransparency * (1 - (percentage / 100))
+)
+
+const getColorValuesInBrackets = (color: string) : string => (
+  color.slice(color.indexOf('(') + 1, color.indexOf(')'))
+)
+
+const getValuesTillLastComma = (color: string) : string => (
+  color.slice(0, color.lastIndexOf(','))
+)
+
+const transparentizeRgb = (rgb: string, percentage: number) : string => (
+  `rgba(${getColorValuesInBrackets(rgb)}, ${getNewTransparency(1, percentage)})`
+)
+
+const getLastColorValue = (color: string) : number => (
+  Number(
+    color.slice(
+      color.lastIndexOf(',') + 1,
+      color.indexOf(')'),
+    ),
+  )
+)
+
+function transparentize(color: string, percentage : number) {
+  if (isHex(color)) {
+    return transparentizeRgb(`${hexToRgb(color)})`, percentage)
+  } else if (isRgb(color)) {
+    return transparentizeRgb(color, percentage)
+  } else if (isRgba(color)) {
+    const currentTransparency = getLastColorValue(color)
+    const rgb = getValuesTillLastComma(getColorValuesInBrackets(color))
+    const newTransparency = getNewTransparency(currentTransparency, percentage)
+
+    return `rgba(${rgb}, ${newTransparency})`
+  } else if (isHsl(color)) {
+    const hslValues = getColorValuesInBrackets(color)
+    const newTransparency = getNewTransparency(1, percentage)
+
+    return `hsla(${hslValues}, ${newTransparency})`
```

Comment:
Right now most of the other functions accept only rgba or hex strings as this is the output from our rgb, rgba, hsl & hsla functions. So even hsla outputs rgba as described here https://github.com/styled-components/polished/issues/44

> - We want to create the smallest possible output. For colors without opacity we use hex e.g. `#FF553A`. If possible only 3 characters `#FFF` instead of `#FFFFFF`. In case of opacity we generate `rgba(255,0,0,0.3)`.

Can you make sure transparentize also exports those? The internal helper contain a couple of helpers that does this conversion.

It might make sense to support `hsl`, `hsla`, `hwb` and `lab` ([all valid CSS](https://drafts.csswg.org/css-color/#the-hwb-notation)) in the future, but I'm not sure. There is a tradeoff. Being open to more inputs means increase code size.